ui-framework-documentation.md

```diff
@@ -76,34 +89,85 @@ sub provide {
 }
 ```
 
-main.pm then calls a scheduled [Test Module](#test-module) that is using
-the Distribution to access its components through Controller layer.
+main.pm calls a scheduled [Test Module](#test-module) using [declarative scheduling](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/declarative-schedule-doc.md)
+(please don't schedule modules in main*.pm directly !) and uses the [Distribution](https://github.com/os-autoinst/os-autoinst-distri-opensuse/tree/master/lib/Distribution)
+to access its components through the [Controller](#controller) layer.
 
 >  Important: Test Module must be inherited from opensusebasetest or one
 >  of its children to have an access to the Distribution.
-
 ```perl
 use parent 'opensusebasetest';
 
 sub run {
     my $partitioner = $testapi::distri->get_partitioner_controller();
```

Comment:
we have omitted the '_controller' part in my methods already, it is kind of redundant. We could pull out of the documentation as well.

Reply:
removed